packages/babel-traverse/src/path/comments.js

```diff
@@ -22,7 +22,7 @@ export function shareCommentsWithSiblings() {
   const hasNext = Boolean(next.node);
   if (hasPrev && !hasNext) {
     prev.addComments("trailing", trailing);
-  } else if (hasNext && !hasPrev) {
+  } else if (hasNext) {
     next.addComments("leading", leading);
   }
```

Comment:
I think the logic we had is wrong, and we should do something like this to avoid loosing comments?

```js
if (hasPrev) {
  prev.addComments("trailing", leading);
} else if (hasNext) {
  next.addComments("leading", leading);
}
if (hasNext) {
  next.addComments("leading", trailing);
} else if (hasPrev) {
  prev.addComments("trailing", trailing);
}
```